README.md
 # beeMo

## LOAD CELL CALIBRATION

1) You need a laptop with VNC installed or a monitor and mouse
3) Open a folder.
4) Go to /home/pi/BeeHiveMonitoring folder
5) Open loadCellCalibration.py
6) Press F5 to Run the script.
7) A python shell will appear and just follow the steps that will be provided. 

## WIFI configuration

1) You need a laptop and a card reader
2) Remove the SD card from the raspberry pi
4) Plug it in your PC/Laptop.
5) WARNING: a format window will pop up when the PC/Laptop read the SD card. JUST IGNORE or CLOSE IT. 
6) DONT FORMAT THE SD CARD!
7) go to boot partition.
8) open the wpa_supplicant.conf file or create a new text file and plug this code:
    <pre> 
          country=ph
          update_config=1
          ctrl_interface=/var/run/wpa_supplicant

          network={
                  scan_ssid=1
                  ssid="yourWIFISSIDHere"
                  psk="yourWIFIPasswordHere"
                  }
    <code>
  9) save the file as wpa_supplicant.conf
  10) copy and paste the file in the boot partition.
  11) remove SD Card from PC and plug it back to the Raspberry Pi.
  12) Check if it is successfully connected by viewing in the monitor or by scanning the IP address within the same network using "Advanced IP Scanner" Software (download from google)


## Pin COnfiguration

* DHT Hive 1 Layer 1  -  GPIO 4
* DHT Hive 1 Layer 2  -  GPIO 17
* DHT Hive 2 Layer 1  -  GPIO 27
* DHT Hive 2 Layer 2  -  GPIO 22
* DHT Hive 3 Layer 1  -  GPIO 10
* DHT Hive 3 Layer 2  -  GPIO 9
* DHT Outsude   -  GPIO 11

* Load Cell Hive 1 - Dout:24, Sck:23
* Load Cell Hive 2 - Dout:8, Sck:25
* Load Cell Hive 3 - Dout:6, Sck:5 

* GSM - RX:GPIO14, TX:GPIO15

* LCD - SCL:GPIO3, SDA:GPIO2



history.txt
{
    "Hive_1": {
        "Humidities": [
            55.099998474121094,
            0
        ],
        "Temperatures": [
            28.600000381469727,
            0
        ],
        "Weight": -0.0010406504065040652
    },
    "Hive_2": {
        "Humidities": [
            55.5,
            59.900001525878906
        ],
        "Temperatures": [
            30.100000381469727,
            29.799999237060547
        ],
        "Weight": 0
    },
    "Hive_3": {
        "Humidities": [
            58.400001525878906,
            0
        ],
        "Temperatures": [
            29.399999618530273,
            0
        ],
        "Weight": 0
    },
    "Outside": {
        "Humidity": 54.400001525878906,
        "Temperature": 29.399999618530273
    },
    "Timestamp": "2019-01-20 22:57:53.072380"
},{'Hive_2': {'Temperatures': [30.0, 29.700000762939453], 'Weight': 0, 'Humidities': [55.900001525878906, 60.0]}, 'Hive_3': {'Temperatures': [14.699999809265137, 0], 'Weight': 0, 'Humidities': [3306.199951171875, 0]}, 'Hive_1': {'Temperatures': [28.600000381469727, 30.399999618530273], 'Weight': -0.0014227642276422765, 'Humidities': [55.599998474121094, 52.5]}, 'Outside': {'Temperature': 29.299999237060547, 'Humidity': 54.599998474121094}, 'Timestamp': '2019-01-20 23:19:45.676636'}

main.py
import DHT
import loadCell
import time
import RPi.GPIO as GPIO
from datetime import datetime
import json
from pprint import pprint
import sys
dht_sensors = [4,17,27,22,10,9,11]

dht = DHT.DHT_SENSOR(dht_sensors)
with open('/home/pi/BeeHiveMonitoring/LoadCell_Calibration_Ratio.json') as f:
                ratio = json.load(f)
                
cell1 = loadCell.Create(24,23,float(ratio["Ratio1"]))
cell2 = loadCell.Create(8,25,float(ratio["Ratio2"]))
cell3 = loadCell.Create(6,5,float(ratio["Ratio3"]))



	try:
    
    while True:
        
    
        temp, humid = dht.getValues()
        weight = []
        weight.append(cell1.Measure())
        weight.append(cell2.Measure())
        weight.append(cell3.Measure())
        for index in range(len(temp)):
            print("{0:.2f}   {1:.2f}".format(temp[index],humid[index]))
             
        print("{0},{1},{2}".format(weight[0],weight[1],weight[2]))
      
        
        try:
            with open('/home/pi/BeeHiveMonitoring/log.json') as f:
                data = json.load(f)

            print(data)
            print("updating values..")

            timestamp = str(datetime.now())
            
            data['Timestamp'] = timestamp
            
            data['Hive_1']['Temperatures'][0] = temp[0]
            data['Hive_1']['Temperatures'][1] = temp[1]
            data['Hive_1']['Humidities'][0] = humid[0]
            data['Hive_1']['Humidities'][1] = humid[1]
            data['Hive_1']['Weight'] = weight[0]
            
            
            data['Hive_2']['Temperatures'][0] = temp[2]
            data['Hive_2']['Temperatures'][1] = temp[3]
            data['Hive_2']['Humidities'][0] = humid[2]
            data['Hive_2']['Humidities'][1] = humid[3]
            data['Hive_2']['Weight'] = weight[1]

            data['Hive_3']['Temperatures'][0] = temp[4]
            data['Hive_3']['Temperatures'][1] = temp[5]
            data['Hive_3']['Humidities'][0] = humid[4]
            data['Hive_3']['Humidities'][1] = humid[5]
            data['Hive_3']['Weight'] = weight[2]

            data['Outside']['Temperature'] = temp[6]
            data['Outside']['Humidity'] = humid[6]
            print("------------------------------")
            pprint(data)
            print("values updated..")
            print("saving..")
            
            with open('/home/pi/BeeHiveMonitoring/log.json','w') as outfile:
                  json.dump(data,outfile,indent=4,sort_keys=True)

            with open('/home/pi/BeeHiveMonitoring/history.json') as f:
                history = json.load(f)

            history['Reports'].append(data)

            with open('/home/pi/BeeHiveMonitoring/history.json','w') as out:
                json.dump(history,out,indent=4,sort_keys = True)
                
            print("Saved!")    
        except KeyboardInterrupt:
            print("Program Interrupted!")
            sys.exit(0)
        
        else:
            print("Done..")
            

       


finally:
    
   
    print("Stopping code..")
    GPIO.cleanup()
      

